src/defrag.c

```diff
@@ -712,6 +739,9 @@ static void defragKey(defragKeysCtx *ctx, robj **elemref) {
         }
         ob = newob;
     }
+    if (key_with_volatile_items) {
+        dbTrackKeyWithVolaItems(db, ob);
+    }
```

Comment:
I think the code should be inside the `if` statement, with - and using the same technique as - `expires_ht`